packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -114,11 +114,24 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
   if (isAppleDevice() && (!hasAriaLabel || isFirefox())) {
     ariaLive = 'off';
   }
+
+  let isPendingProps = isPending ? {
+    onClick: (e) => {
+      if (e.currentTarget instanceof HTMLButtonElement && e.currentTarget.type === 'submit') {
```

Comment:
i think we can always prevent the default as I imagine we'd need to do it for type === 'reset' too.

Can you think of a reason we'd allow it?

I think this is a fine place for the check, we'll eventually move pending logic into RAC, so we can always reevaluate then.

Reply:
Thanks for the review!

Yeah I think it's safe to prevent default as long as the element is `<button>` since it makes sense to prevent the default behaviors for `type=submit` and `type=reset`, and `type=button` actually doesn't have a default behavior, [it seems](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#type).